models.py

```diff
@@ -149,25 +151,13 @@ def generate_compiled_segmentation_model(model_name, model_parameters, num_class
 
     # strategy = tf.distribute.MirroredStrategy()
     # with strategy.scope():
-    if model_name == "Unet":
-        if model_parameters['backbone_name'] in Unet_backbones:
-            model = Unet(classes=num_classes, **model_parameters)
-        else:
-            raise NameError("Error, model and backbone are not compatible.")
-    elif model_name == "FPN":
-        if model_parameters['backbone_name'] in FPN_backbones:
-            model = FPN(classes=num_classes, **model_parameters)
-        else:
-            raise NameError("Error, model and backbone are not compatible.")
-    elif model_name == "Linknet":
-        if model_parameters['backbone_name'] in Linknet_backbones:
-            model = Linknet(classes=num_classes, **model_parameters)
-        else:
-            raise NameError("Error, model and backbone are not compatible.")
+
```

Comment:
add high-level if-else to make sure `[model_name]` is in `models_dict`, proceed if true but throw helpful error message if choosing a currently unsupported option